mavros_extras/src/plugins/distance_sensor.cpp

```diff
@@ -242,7 +296,7 @@ DistanceSensorItem::Ptr DistanceSensorItem::create_item(DistanceSensorPlugin *ow
 		}
 
 		// optional
-		pnh.param("covariance", p->covariance, 0);
+		if (pnh.getParam("covariance", p->covariance) > 0) p->cov_is_def = true;
```

Comment:
`getParam() -> bool`, so this is wrong.
Leave here `pnh.param("covariance", p->covariance, 0);` and use value of covariance to check defined or not.